ata_api/main.py

```diff
@@ -1,47 +1,82 @@
-from os import getenv
-from random import randint
+from uuid import UUID
 
 from ata_db_models.helpers import get_conn_string
-from ata_db_models.models import Prescription
+from ata_db_models.models import Group, UserGroup
 from fastapi import FastAPI
 from mangum import Mangum
-from sqlmodel import Session, create_engine, select
+from pydantic import BaseModel
+from sqlmodel import Session, create_engine, select, text
 
 app = FastAPI()
 engine = create_engine(url=get_conn_string())
-POLICY = getenv("POLICY", "RCT")
+
+
+class PrescriptionResponse(BaseModel):
+    group: Group
+    value: int
 
 
 @app.get("/")
 def get_root() -> object:
     return {"message": "This is the root endpoint for the AtA API."}
 
 
-@app.get("/prescription/{site_name}/{user_id}", response_model=int)
-def read_prescription(site_name: str, user_id: str) -> int:
-    if POLICY == "RCT":
-        return fair_coin_toss()
-    else:
-        return get_prescription_from_db(site_name=site_name, user_id=user_id)
-
-
-def fair_coin_toss() -> int:
-    return randint(0, 1)
-
-
-def ignore_existing_subscribers() -> bool:
-    # todo
-    ...
-
-
-def get_prescription_from_db(site_name: str, user_id: str) -> int:
+@app.get("/prescription/{site_name}/{user_id}", response_model=PrescriptionResponse)
+def read_prescription(site_name: str, user_id: str) -> PrescriptionResponse:
     with Session(engine) as session:
-        prescription = session.exec(
-            select(Prescription).where(Prescription.user_id == user_id, Prescription.site_name == site_name)
+        # query db for user/site group
+        usergroup = session.exec(
+            select(UserGroup).where(UserGroup.user_id == user_id, UserGroup.site_name == site_name)
         ).first()
-        if prescription:
-            return int(prescription.prescribe)
-        return 0
+        # if no row for the user/site, create it
+        if not usergroup:
+            usergroup = UserGroup(user_id=UUID(user_id), site_name=site_name)
+            session.add(usergroup)
+            session.commit()
+
+        if usergroup.group == Group.A:
+            # show CTA
+            return PrescriptionResponse(group=Group.A, value=1)
+        elif usergroup.group == Group.B:
```

Comment:
📌 I think we'll need to grant `dev_api` and `prod_api` SELECT privilege to the event table.